### PR TITLE
Add: New package JSON hash added

### PR DESCRIPTION
**Pull request by ThulinaWickramasinghe**
This package is intended to be used for hashing a provided JS object in both Node.js and browser(eg-: for Next.js). Users will be provided an option to sort the JS object since order of properties is not guaranteed by JSON.stringify. This sorting will be done using this [package](https://github.com/ljharb/json-stable-stringify) which will be imported dynamically. In the case of a string being passed as a parameter to the algorithm, I've kept sorting optional.

On the browser side hashing will be done using the inbuild [SubteCrypto Web API](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto) which only provides the following algorithms for creating a hash.

- SHA-1
- SHA-256
- SHA-384
- SHA-512

In node.js we use the inbuilt [crypto module](https://nodejs.org/api/crypto.html), Even though node.js crpto module provides more algorithms than the ones available in the browser to maintain cross platform compatibility (browser and node.js), I've limited the available algorithms only the ones available in browser.

### CODE REVIEW

File: packages/json-hash/src/algorthms.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  "SHA-384": "sha384",
+  "SHA-512": "sha512"
+};
+
+export const getNodeJSAlgoName = (algorithm) => {
+  return algos[algorithm];
+};
+
+export const isAvailableAlgo = (algorithm) => {
+  if (Object.prototype.hasOwnProperty.call(algos, algorithm)) {
```
Comment (Akalanka47000) [inline packages/json-hash/src/algorthms.js:13]: We can use algos[algorithm] here instead right, will be more readable

File: packages/json-hash/test/index.test.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
+import { cloneDeep } from "lodash";
+import { computeHash } from "../src";
+import { getNodeJSAlgoName, isAvailableAlgo } from "../src/algorthms";
+
+describe("json-hash-node-js", () => {
+  const objectToHash = { name: "Siri", age: 30, country: "Sri Lanka" };
+
+  beforeAll(() => {
+    global.unit_tests_running = true;
```
Comment (Akalanka47000) [inline packages/json-hash/test/index.test.js:9]: I think we won't need this, this flag was used to handle a few edge cases of tests in a couple of other packages

**Review by Akalanka47000 — CHANGES_REQUESTED** [review]

File: packages/json-hash/src/algorthms.js (hunk 1 above)
Comment (ThulinaWickramasinghe) [reply packages/json-hash/src/algorthms.js:13]: Will do

File: packages/json-hash/test/index.test.js (hunk 2 above)
Comment (ThulinaWickramasinghe) [reply packages/json-hash/test/index.test.js:9]: Will remove

Comment (ThulinaWickramasinghe) [reply packages/json-hash/test/index.test.js]: Done

File: packages/json-hash/src/algorthms.js (hunk 1 above)
Comment (ThulinaWickramasinghe) [reply packages/json-hash/src/algorthms.js]: Done
